core/src/main/java/com/sequenceiq/cloudbreak/service/eventbus/CloudResourcePersisterService.java

```diff
@@ -31,7 +32,7 @@
     @Inject
     private ResourceService resourceService;
 
-    @Inject
+    @PersistenceContext
```

Comment:
Please add a detailed commit message that why we need to change it, what problem is caused by the old solution. I just want to avoid that situation that a year from now somebody would accidentally/intentionally write it back.

Please also double-check whether we could eliminate the following lines from DatabaseConfig java files. 

   @Bean
    public EntityManager entityManager(EntityManagerFactory entityManagerFactory) {
        return entityManagerFactory.createEntityManager();
    }

https://github.com/hortonworks/cloudbreak/blob/fe089b5bf16558e4478f3c71165cc485d8cdc0e3/autoscale/src/main/java/com/sequenceiq/periscope/config/DatabaseConfig.java#L119

Reply:
Detailed commit message and java comment added. 
It seems the referred DatabaseConfig lines can be removed, applications are able to start without these lines.